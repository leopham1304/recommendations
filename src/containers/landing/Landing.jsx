import React, { memo } from "react";
import { useHandleButtonClicked } from "./Landing.utils";
import "./Landing.scss";

const Landing = () => {
  const handleButtonClicked = useHandleButtonClicked();

  return (
    <div className="landing-container">
      <div className="introduction-wrapper">
        <h1>Hello, welcome to Vietnam!</h1>
        <div className="introduction-text">
          <p>Vietnam is a country divided into 58 provinces</p>
          <p>There are many heavenly breathtaking landscapes to explore</p>
        </div>
        <button onClick={handleButtonClicked} className="landing-button">
          Fly me to the <strong> Moon </strong>
        </button>
      </div>
    </div>
  );
};

export default memo(Landing);
