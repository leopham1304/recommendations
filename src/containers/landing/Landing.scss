.landing-container {
  text-align: center;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: url(../../assets/images/sondoong.jpg);
  color: #fff;
  height: 100%;

  h1 {
    margin: 0;
    padding: 0;
    margin-bottom: 1vw;
  }

  .introduction-wrapper {
    top: 15vw;
    position: relative;
  }

  .introduction-text {
    font-weight: 500;
    margin: 0;
    margin-bottom: 1vw;
  }

  .landing-button {
    background-color: #699db6;
    border-color: rgba(0, 0, 0, 0.3);
    text-shadow: 0 1px 0 rgba(0, 0, 0, 0.5);
    color: #fff;
    cursor: pointer;
    padding: 5px;
  }

  .landing-button:hover {
    background-color: #4f87a2;
    border-color: rgba(0, 0, 0, 0.5);
  }

  .landing-button:active {
    background-color: #3c677b;
    border-color: rgba(0, 0, 0, 0.9);
  }
}
