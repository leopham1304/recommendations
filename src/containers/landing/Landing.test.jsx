import React from "react";
import { Router } from "react-router-dom";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Landing from "./Landing";

describe("Landing", () => {
  const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
  const wrapper = mount(
    <Router history={historyMock}>
      <Landing />
    </Router>
  );

  it("should render the Landing correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("handleButtonClicked binds properly", () => {
    wrapper.find("button").simulate("click");
    expect(historyMock.push.mock.calls[0][0]).toEqual("/home");
  });
});
