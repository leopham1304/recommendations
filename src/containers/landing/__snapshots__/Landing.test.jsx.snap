// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render the Landing correctly 1`] = `
<Router
  history={
    Object {
      "listen": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "location": Object {},
      "push": [MockFunction],
    }
  }
>
  <Memo(Landing)>
    <div
      className="landing-container"
    >
      <div
        className="introduction-wrapper"
      >
        <h1>
          Hello, welcome to Vietnam!
        </h1>
        <div
          className="introduction-text"
        >
          <p>
            Vietnam is a country divided into 58 provinces
          </p>
          <p>
            There are many heavenly breathtaking landscapes to explore
          </p>
        </div>
        <button
          className="landing-button"
          onClick={[Function]}
        >
          Fly me to the 
          <strong>
             Moon 
          </strong>
        </button>
      </div>
    </div>
  </Memo(Landing)>
</Router>
`;
